src/v/cluster/tx_coordinator_mapper.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "cluster/metadata_cache.h"
+#include "cluster/partition_manager.h"
+#include "cluster/tm_tx_hash_ranges.h"
+#include "hashing/murmur.h"
+#include "kafka/protocol/types.h"
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "model/namespace.h"
+#include "seastar/core/sharded.hh"
+#include "seastarx.h"
+
+#include <seastar/core/reactor.hh>
+
+namespace cluster {
+
+inline model::partition_id get_partition_from_default_distribution(
+  uint32_t tx_id_hash, int32_t partitions_amount) {
+    uint32_t default_partition_range_size = get_default_range_size(
+      partitions_amount);
+    int32_t partition = int32_t(tx_id_hash / default_partition_range_size);
+
+    // Last partition in default distibuiton can have bigger range
+    if (partition >= partitions_amount) {
+        return model::partition_id(partitions_amount - 1);
+    }
```

Comment:
maybe account for this case in the original division?

Reply:
what do you mean?

Reply:
oh it just seemed like you could do like `m / (n+1)` or something to get the same effect. i didn't actually confirm that's possible.

Reply:
problem here `m / (n+1)` is when we will have only one partition
We will have overflow because `default_partition_range_size`  will be equal to UINT32_MAX
Or we will need to make one more if with checking that we have only one partition or something else

Reply:
ahh ok thanks makes sense